src/plugins/intel_cpu/src/nodes/segment_max.cpp

```diff
@@ -0,0 +1,120 @@
+// Copyright (C) 2018-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "segment_max.h"
+
+#include "openvino/op/segment_max.hpp"
+#include "openvino/reference/segment_max.hpp"
+
+namespace ov::intel_cpu::node {
+SegmentMax::SegmentMax(const std::shared_ptr<ov::Node>& op, const GraphContext::CPtr& context)
+    : Node(op, context, NgraphShapeInferFactory(op)) {
+    std::string errorMessage;
+    if (!isSupportedOperation(op, errorMessage)) {
+        OPENVINO_THROW_NOT_IMPLEMENTED(errorMessage);
+    }
+    fillMode = ov::as_type_ptr<const ov::op::v16::SegmentMax>(op)->get_fill_mode();
+}
+
+bool SegmentMax::isSupportedOperation(const std::shared_ptr<const ov::Node>& op, std::string& errorMessage) noexcept {
+    try {
+        if (!ov::is_type<ov::op::v16::SegmentMax>(op)) {
+            errorMessage = "Only opset16 SegmentMax operation is supported";
+            return false;
+        }
+    } catch (...) {
+        return false;
+    }
+    return true;
+}
+
+void SegmentMax::getSupportedDescriptors() {
+    // Validation is already done in the ov::opset16::SegmentMax
+}
+
+void SegmentMax::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty()) {
+        return;
+    }
+    ov::element::Type dataPrecision = getOriginalInputPrecisionAtPort(0);
+    ov::element::Type segmentIdsPrecision = getOriginalInputPrecisionAtPort(1);
+
+    if (getOriginalInputsNumber() == 2) {
+        addSupportedPrimDesc({{LayoutType::ncsp, dataPrecision}, {LayoutType::ncsp, segmentIdsPrecision}},
+                             {{LayoutType::ncsp, dataPrecision}},
+                             impl_desc_type::ref);
+    } else {
+        ov::element::Type numSegmentsPrecision = getOriginalInputPrecisionAtPort(2);
+        addSupportedPrimDesc({{LayoutType::ncsp, dataPrecision},
+                              {LayoutType::ncsp, segmentIdsPrecision},
+                              {LayoutType::ncsp, numSegmentsPrecision}},
+                             {{LayoutType::ncsp, dataPrecision}},
+                             impl_desc_type::ref);
+    }
+}
+
+bool SegmentMax::created() const {
+    return getType() == Type::SegmentMax;
+}
+
+bool SegmentMax::needPrepareParams() const {
+    return false;
+}
+
+void SegmentMax::executeDynamicImpl(const dnnl::stream& strm) {
+    execute(strm);
+}
+
+template <class T, class T_idx>
+void SegmentMax::executeImpl() {
+    const auto& data_shape = getSrcMemoryAtPort(0)->getStaticDims();
+    const auto& output_shape = getDstMemoryAtPort(0)->getShape().getStaticDims();
```

Comment:
As output shape can be related to the `num_segments` input (if provided). Consider implementation of some additional logic and override the "needShapeInfer()" from CPU Node class.

Reply:
Done.